app/models/spree/gateway/amazon.rb

```diff
@@ -135,5 +146,11 @@ def void(response_code, gateway_options)
     def load_amazon_mws(reference)
       @mws ||= AmazonMws.new(reference, gateway: self)
     end
+
+    # A random string of lowercase alphanumeric characters (i.e. "base 36")
+    def random_suffix
```

Comment:
Ah I see your point, I guess we don't need much randomness considering the payment number is already unique, but one can never be too safe I guess